### DISCUSSION TITLE
Auth Webhook "request" is null

### DISCUSSION THREAD

**Discussion by dereky565** [hasura/graphql-engine#10623] [Question] [answered]
This is the payload content my Auth Webhook receives when I am making a graphql request:
```
{
headers: {
Accept: "*/*",
Accept-Encoding: "gzip, deflate, br, zstd",
Accept-Language: "en-US,en;q=0.5",
Authorization: "*",
Connection: "keep-alive",
Content-Length: "306",
DNT: "1",
Host: "localhost:8080",
Origin: "http://localhost:8080",
Priority: "u=0",
Referer: "http://localhost:8080/console",
Sec-Fetch-Dest: "empty",
Sec-Fetch-Mode: "cors",
Sec-Fetch-Site: "same-origin",
User-Agent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0",
content-type: "application/json",
x-request-id: "39d3dedb-4cfd-438e-b8a5-5107aa9acfa7"
},
request: {
operationName: "MyQuery",
query: "query MyQuery {\n  stauserrole {\n    activate\n    deactivate\n    defappid\n    lastupdate\n    parentlglentityid\n    sortseq\n    status\n    updateuserid\n    userrolecode\n    userrolehome\n    userrolename\n    usertypeid\n    version\n  }\n}\n"
}
}
```

But this is the payload content my Auth Webhook received when I make a request to a RESTified Hasura Endpoint:
(Request field is null in POST mode for Auth Webhook when making requests to RESTified endpoints on Hasura.)
```
{
headers: {
Accept: "*/*",
Accept-Encoding: "gzip, deflate, br, zstd",
Accept-Language: "en-US,en;q=0.5",
Authorization: "*",
Connection: "keep-alive",
Content-Length: "31",
DNT: "1",
Host: "localhost:8080",
Origin: "http://localhost:8080",
Priority: "u=0",
Referer: "http://localhost:8080/console/api/rest/details/Create%20User%20Role",
Sec-Fetch-Dest: "empty",
Sec-Fetch-Mode: "cors",
Sec-Fetch-Site: "same-origin",
User-Agent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0",
content-type: "application/json",
x-request-id: "7521475a-5b46-4197-98e1-c7a7c8970815"
},
request: null
}
```

Is there a possibility that I am missing some configuration or is this the current behavior? The Hasura version used is also v2.26.0

Edit: I also set my HASURA_GRAPHQL_AUTH_HOOK_MODE to POST in my docker-compose and HASURA_GRAPHQL_AUTH_HOOK_SEND_REQUEST_BODY to true.

**Comment by dereky565** [comment]
Hi, I also tried updating my Hasura to v2.45.1 and the behavior remains the same wherein only graphql requests populate the "request" field while the RESTified Endpoint requests "request" field is still null.

**Comment by robertjdominguez** [comment]
@dereky565 — sorry for the delay. FWIW, this is stumping me, too 😅 

Let me reach out to someone who was on the team that built this feature and see if they can drop some knowledge for us all 🙏

**Comment by daniel-chambers** [comment, marked as answer]
Unfortunately the `request` field is not set when using a restified endpoint. The auth hook is invoked before we actually process the request and look up the GraphQL stored for that rest endpoint, so it's not available to be sent to the auth hook. This is unlike normal GraphQL requests where the query text is in the incoming request body, so we've got it to send to the auth hook (hence why it's in the "request" property). For a restified endpoint, the graphql query text is not in the http request, so the "request" property ends up being null.

**Reply by dereky565** [reply]
Hi, thank you for this! So this really is the expected behavior after all.
